specification/dartLangSpec.tex

```diff
@@ -1267,8 +1267,10 @@ \section{Errors and Warnings}
 are situations that do not preclude execution,
 but which are unlikely to be intended,
 and likely to cause bugs or inconveniences.
-A static warning must be reported by a Dart compiler
-before the associated code is executed.
+A Dart compiler is free to report some, all,
+or none of the specified static warnings before executing the associated code.
+A Dart compiler may also report additional warnings
```

Comment:
I'd consider making it "... may also choose to report ...". Just to emphasize that it's a *choice*, and it's not because there is some mandated list of warnings outside of this specification.

Reply:
+1: https://github.com/dart-lang/language/pull/3584